support/tech-resources/enabling-https-engagement-tracking-on-sparkpost.md

```diff
@@ -267,24 +286,87 @@ If you use AWS CloudFront to enable HTTPS engagement tracking, by default, Cloud
+
+## <a name="fastly-cert"></a> Issue a certificate with Fastly
+
+1. Select the "HTTPS and network" tab, then "Get Started".
+
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-cert1.png)
+
+
+1. Enter your tracking domain. Let's Encrypt certificates are free, and can be auto-renewed by Fastly, via an additional CNAME record that you will need to create with your DNS provider.
+
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-cert2.png)
+
+    Other options are to use GlobalSign, or to upload your own private key & certificate.
+
+1. For Let's Encrypt option: copy the information shown and create a CNAME record in your DNS provider's account.
+
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-cert-letsencrypt-cname.png)
 
-    Set the options as follows:
+1. After you create the CNAME, Fastly requests the certificate.
 
-    Server address and port: For SparkPost US, this would be `spgo.io :  443`; for SparkPost EU, this would be `eu.spgo.io : 443`. For PowerMTA+Signals, refer to your PowerMTA User Guide.
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-cert-letsencrypt2.png)
 
-    Domain: Your click tracking domain, e.g. `click.business.com`
-    Description: SparkPost (or whatever you like!)
+    After a short time, you should see
+
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-cert-letsencrypt3.png)
+
+1. Select "More Details .." and look for "CNAME records". This is the address the Fastly will use to serve your incoming requests.
+
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-service-cname.png)
+
+1. Create the CNAME record within your DNS service (this will be specific to your provider). If you have a TTL (time to live) field, we suggest to set this to 1 hour.
+
+    ![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-cname.png)
+
+    _Example DNS provider CNAME setup_
+
+    You can verify that the routing is successful using `ping` on your created record.
 
 1. Follow [these steps](#switch-to-secure) to update and verify your tracking domain.
+
+Fastly keeps previous versions of your configuration, and can show the "diff" between them. You can also set up advanced routing rules using the VCL language, and monitor statistics on served requests.
+
+![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-stats.png)
+
+
 ---
 
-## <a name = "switch-to-secure"></a> Switch tracking domain to secure, and validate
+## <a name="ttl"></a> Cache Time To Live (TTL) settings
+
+CDNs apply caching, with a "Time to Live" (TTL) for each unique request URL. When a request is first fetched, it is cached. Within the TTL, later requests *may* be served to the client from the CDN cache, without touching the SparkPost endpoint. The client is redirected to the landing page as usual, but a side-effect is that SparkPost does not record the repeat opens/clicks.
```

Comment:
@ndurant-sp @ischer For opens, we return `Cache-Control: no-cache, max-age=0`. For clicks, we do not return a Cache-Control header at all. This appears to be true for both Momentum and the new ET service.

Given what Steve found during the preparation of these changes, I think we should consider returning some kind of TTL in a Cache-Control for clicks too.

Both Cloudflare and AWS say they will cache 302 redirects:

 * https://support.cloudflare.com/hc/en-us/articles/200172516-Are-301-and-302-redirects-cached-by-CloudFlare-
 * https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html

Interestingly, the now-obsolete old HTTP standard recommended against caching 302s (SHOULD) but the new RFC does not. See old one https://tools.ietf.org/html/rfc2616#section-10.3.3 and new one https://tools.ietf.org/html/rfc7231#section-6.4.3